crates/bevy_ecs/src/world/mod.rs

```diff
@@ -777,7 +777,7 @@ impl World {
     /// Note that any resource with the `Default` trait automatically implements `FromWorld`,
     /// and those default values will be here instead.
     #[inline]
-    pub fn init_resource<R: Resource + FromWorld>(&mut self) {
+    pub fn init_resource<R: Resource + FromWorld>(&mut self) -> ComponentId {
```

Comment:
You should make the same change for `init_non_send_resource`.

Reply:
Ah, didn't notice that function.